complex input1
complex input2
---
complex output
